### PR TITLE
ASoC: codecs: add support for ES8326

### PR DESCRIPTION
**Pull request by yangxiaohua2009**
The ES8326 codec is not compatible with ES8316 and requires a dedicated driver.

Signed-off-by: David Yang <yangxiaohua@everest-semi.com>

### CODE REVIEW

**Comment by sofci** [comment]
Can one of the admins verify this patch?
> reply **test&nbsp;this&nbsp;please** to run this test once

**Comment by plbossart** [comment]
test this please

**Comment by plbossart** [comment]
@yangxiaohua2009 you need to rebase this branch on topic/sof-dev, there should only be one commit in this PR

File: sound/soc/codecs/es8326.c [hunk 1]
```diff
@@ -0,0 +1,819 @@
...
+	es8326->i2c = i2c;
+	mutex_init(&es8326->lock);
+	es8326->regmap = devm_regmap_init_i2c(i2c, &es8326_regmap_config);
+	if (IS_ERR(es8326->regmap)) {
+		ret = PTR_ERR(es8326->regmap);
+		dev_err(&i2c->dev, "Failed to init regmap: %d\n", ret);
+		return ret;
+	}
+
+	es8326->mclk = devm_clk_get_optional(component->dev, "mclk");
```
Comment (plbossart) [inline sound/soc/codecs/es8326.c:756]: this won't compile, see GitHub actions.

**Review by plbossart — CHANGES_REQUESTED** [review]
need to rebase and fix compilation issues.

**Pull request closed without merging**